zk/conn.go

```
@@ -984,6 +968,9 @@ func (c *Conn) Children(path string) ([]string, *Stat, error) {
 
 	res := &getChildren2Response{}
 	_, err := c.request(opGetChildren2, &getChildren2Request{Path: path, Watch: false}, res, nil)
+	if err != nil && err == ErrConnectionClosed {
+		return nil, nil, err
```

not sure why we don't want to return an error to the caller if the connection is closed.

could you explain?

---

Yeah im not super sure on this one as it comes from the mesos fork. I will consider this and look more into why this was done. it might be unwanted.

---

So after looking the the comments and commit messages it seems the request if we fail in a certain was will not be able to handle returning the desired struct values or fields since an error has occurred. I will actually remove this with the notion this is an issue that we can solve later with better send and receive loop control. Even more i think i'm just going to cheery pick the once change from mesospshere being the Close once change and drop the rest. this seems like the wrong solution to a problem we have not observed. If there were tests this would be easier to argue but since nothing is there i will just remove all these changes.

---

updated with removing commits around this connection error handling. the changes are more concise and do not change the returning of errors. I think the racy comments around queuing requests from the commits were valid but we can solve this with doing a better coordination of send/recv routines.